invertedai/logs/diagnostics.py

```diff
@@ -0,0 +1,140 @@
+        self,
+        debug_log_path: str
+    ):
+        """
+        A user-side tool that examines a debug log checking for common implementation mistakes 
+        and provides feedback.
+
+        Parameters
+        ----------
+        debug_log_path:
+            The full path to the debug log file to be loaded and analyzed.
+        """
+
+        self.debug_log_path = debug_log_path
+        self.log_data = None
+        with open(debug_log_path) as json_file:
+            self.log_data = json.load(json_file)
+
+    def full_diagnostic_test(self):
+        """
+        Main access point for this class. This function runs all available diagnostic tests and prints 
+        human-readable feedback for the benefit of a user to fix issues in their integration.
+        """
+
+        drive_state_equivalence_messages = self._check_drive_response_equivalence()
+
+        for msg in drive_state_equivalence_messages:
+            print(f"Diagnostic Information: {msg}")
+
+        print(f"Finished diagnostic analysis.")
+
+    def _check_drive_response_equivalence(self):
+        diagnostic_messages = []
+
+        is_equal_agent_states = {"is_equal":[],"agents_equal":[]}
+        is_any_agent_states_issue = False
+        is_equal_recurrent_states = {"is_equal":[],"agents_equal":[]}
+        is_any_recurrent_states_issue
+
+        req_groupings, res_groupings = self._get_all_drive_agents(log_data = self.log_data)
+
+        for req_dict, res_dict in zip(req_groupings["agent_states"][1:],res_groupings["agent_states"][:-1]):
+            states_equal, is_state_equal = self._check_states_equal(req_dict,res_dict)
+            is_equal_agent_states["agents_equal"].append(states_equal)
+            is_equal_agent_states["is_equal"].append(is_state_equal)
+        for req_dict, res_dict in zip(req_groupings["recurrent_states"][1:],res_groupings["recurrent_states"][:-1]):
+            states_equal, is_state_equal = self._check_states_equal(req_dict,res_dict)
+            is_equal_recurrent_states["agents_equal"].append(states_equal)
+            is_equal_recurrent_states["is_equal"].append(is_state_equal)
+
+        if not all(is_equal_agent_states["is_equal"]):
+            res_states = [f"({j}:{is_equal_agent_states["agents_equal"][j]})" for j, val in enumerate(is_equal_agent_states["is_equal"]) if not val]
+            diagnostic_messages.append(f"Potential agent state index error detected for (Time step:[Agent Indexes]): {res_states}")
+
+        if not all(is_equal_recurrent_states["is_equal"]):
+            res_states = [f"({j}:{is_equal_recurrent_states["agents_equal"][j]})" for j, val in enumerate(is_equal_recurrent_states["is_equal"]) if not val]
+            diagnostic_messages.append(f"Potential recurrent state index error detected for (Time step:[Agent Indexes]): {res_states}")
+
+        return diagnostic_messages
+
+    def _get_all_drive_agents(
+        self,
+        log_data: Dict
+    ):
+        STATE_SIGDIG = 2
```

Comment:
Naming wise since you are using it with round(), i believe they are number of decimal places not significant digit